src/drawer.canvasentry.js

```diff
@@ -103,19 +105,21 @@ export default class CanvasEntry {
     updateDimensions(elementWidth, totalWidth, width, height) {
         // where the canvas starts and ends in the waveform, represented as a
         // decimal between 0 and 1
-        this.start = this.wave.offsetLeft / totalWidth || 0;
+        this.start = this.wave[this.orientation.attrFor('offsetLeft')] / totalWidth || 0;
```

Comment:
I don't like this `attrFor` approach and would definitely prefer to keep the old object notation, e.g. `wave.offsetLeft`.

Reply:
TBH, I don't especially like it either, but I'm not sure how to do what you propose. We need some kind of shim to translate attributes such as this one for vertically oriented waveforms; can you suggest a way to do it more transparently, per your request? It seems like it would require overriding `wave` with some kind of wrapper, but this would be necessary for every other object for which we access the DOM attributes as well. Is that what you had in mind?

Reply:
yea let's brainstorm about that. paging @sundayz @marizuccara

Reply:
Perhaps a [Proxy object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) would do the trick? I'll take a stab at it the next chance I have to work on this.

Reply:
I just [checked in](https://github.com/katspaugh/wavesurfer.js/pull/2195/commits/117bbd4bfcfb41074866c01f45c6f89d61a9529e) a rough implementation of this but [Proxy support](https://caniuse.com/?search=proxy) may be prohibitively low. Babel flags it as undefined. I'm not sure what other options are but I may try a manual wrapper of some kind next.

Reply:
huh, Proxy is supported in 99% of the browsers isn' it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy#browser_compatibility

Reply:
Support seems decent in modern browsers but you would lose it for some older ones. I'm not sure if that's a concern. I would like to hash out a solution to this piece of things before addressing your other comments (which I'm happy to do) in case this turns out to be a hard blocker. Here's the options I see:
1. Implement this using Proxy objects, drop support entirely for older browsers as necessary. Not sure what to do about Babel flagging it in this case though.
2. Implement using Proxy objects but with a warning that vertical rendering is not supported in older browsers. Do a check in `orientation.js` whether `Proxy` is defined and, if not, make its functionality a no-op. Not sure if this will pass the linter.
3. Add a [Babel plugin](https://www.npmjs.com/package/babel-plugin-proxy) for a polyfill. It's 5 years old and warns "It is not suitable for production environments because performance impact is huge," so it may not be a _great_ option, but it is an option.
4. Something I'm missing??? I don't know if there's a way to do this cleanly without Proxy.

Reply:
> 1. drop support entirely for older browsers as necessary.

What browsers are we talking about exactly? and there's always a polyfill right?

> Do a check in orientation.js whether Proxy is defined and, if not, make its functionality a no-op. Not sure if this will pass the linter.

Sounds good.

Reply:
ps. https://github.com/katspaugh/wavesurfer.js/blob/master/.eslintrc.js probably needs `Proxy` as well..?

Reply:
Pretty sure IE11 is the only browser that doesn't have Proxy support and can't be polyfilled

Reply:
> I just [checked in](https://github.com/katspaugh/wavesurfer.js/pull/2195/commits/117bbd4bfcfb41074866c01f45c6f89d61a9529e) a rough implementation of this but [Proxy support](https://caniuse.com/?search=proxy) may be prohibitively low. Babel flags it as undefined. I'm not sure what other options are but I may try a manual wrapper of some kind next.
> Add a [Babel plugin](https://www.npmjs.com/package/babel-plugin-proxy) for a polyfill. It's 5 years old and warns "It is not suitable for production environments because performance impact is huge," so it may not be a great option, but it is an option.

I'm trying to reproduce this `Babel flags it as undefined` because babel-proxy-plugin `is 5 years old and warns "It is not suitable for production environments because performance impact is huge` which is problematic of course, stupid I missed that.

Reply:
> Pretty sure IE11 is the only browser that doesn't have Proxy support and can't be polyfilled

Good point. I removed it and compilation doesn't fail. Let's see if the tests succeed as well and I'll remove it. See #2568